types/firefox-webext-browser/index.d.ts

```diff
@@ -0,0 +1,3128 @@
+    }
+
+    class ExtensionSidebarPane {
+        setHeight(height: string): void;
+
+        setExpression(expression: string, rootTitle?: string): Promise<void>;
+
+        setObject(jsonObject: string, rootTitle?: string): Promise<void>;
+
+        setPage(path: string): void;
+
+        onShown: EventListener<(window: object/*global*/) => void>;
+        onHidden: EventListener<() => void>;
+    }
+
+    class Button {
+        update(tooltipText?: string, disabled?: boolean): void;
+        update(disabled?: boolean): void;
+        update(iconPath: string, tooltipText: string, disabled?: boolean): void;
+
+        onClicked: EventListener<() => void>;
+    }
+
+    /* devtools.panels properties */
+    const elements: ElementsPanel;
+
+    const sources: SourcesPanel;
+
+    const themeName: string;
+
+    /* devtools.panels functions */
+    function create(title: string, iconPath: string, pagePath: string): Promise<ExtensionPanel>;
+
+    function setOpenResourceHandler(): Promise<devtools.inspectedWindow.Resource>;
+
+    function openResource(url: string, lineNumber: number): Promise<void>;
+
+    /* devtools.panels events */
+    const onThemeChanged: EventListener<(themeName: string) => void>;
+}
+
+declare namespace browser.find {
+    /* find functions */
+    function find(queryphrase: string, params?: {
+        tabId?: number;
+        caseSensitive?: boolean;
+        entireWord?: boolean;
+        includeRectData?: boolean;
+        includeRangeData?: boolean;
+    }): void;
+
+    function highlightResults(params?: {
+        rangeIndex?: number;
+        tabId?: number;
+        noScroll?: boolean;
+    }): void;
+
+    function removeHighlighting(tabId?: number): void;
+}
+
+declare namespace browser.geckoProfiler {
+    /* geckoProfiler types */
+    enum ProfilerFeature {
+        java = "java",
+        js = "js",
+        leaf = "leaf",
+        mainthreadio = "mainthreadio",
+        memory = "memory",
+        privacy = "privacy",
+        restyle = "restyle",
+        stackwalk = "stackwalk",
+        tasktracer = "tasktracer",
+        threads = "threads"
+    }
+
+    /* geckoProfiler functions */
+    function start(settings: {
+        bufferSize: number;
+        interval: number;
+        features: ProfilerFeature[];
+        threads?: string[];
+    }): void;
+
+    function stop(): void;
+
+    function pause(): void;
+
+    function resume(): void;
+
+    function getProfile(): void;
+
+    function getProfileAsArrayBuffer(): void;
+
+    function getSymbols(debugName: string, breakpadId: string): void;
+
+    /* geckoProfiler events */
+    const onRunning: EventListener<(isRunning: boolean) => void>;
+}
+
+declare namespace browser.history {
+    /* history types */
+    enum TransitionType {
+        link = "link",
+        typed = "typed",
+        auto_bookmark = "auto_bookmark",
+        auto_subframe = "auto_subframe",
+        manual_subframe = "manual_subfram
```

Comment:
`browser.menusInternal` is not not exposed to the extension code as a valid API namespace, but the OnClickData type  defined here seems to be needed because it is used in the `browser.menu.update` API method

(to be fair, the two menu.OnClickData and menusInternal.OnClickData types seems to be similar but not in sync in the Firefox API schema, I'm going to take a deeper look to how we are using them internally).

Reply:
So would an okay solution for now be to maybe rename `menusInternal` to `_menusInternal`?